### PR TITLE
Add a text-based calendar that parses .ics from our NextCloud calendar 

### LINKED ISSUE #4: Human-readable front page calendar
Chatted with finn about this in person at some point, response to #3

would be cool if we could reproduce the former state of the front page calendar but programatically

image

https://github.com/jkbrzt/rrule
https://github.com/aditosoftware/rrule-parser

### PR DESCRIPTION
**Pull request by leafcompost**
This uses ical.js to load a calendar from our NextCloud .ics export and turn it into text. As future work, the JavaScript could use some cleanup (in part by reusing third party libraries to do ical rrule->English language conversion), but this seems to work pretty well for the current events on the agenda.

text-based calendar using html ul element

Thanks to @thefinn93 for helping with reverse proxying the ics file.

Fixes #4.

### CODE REVIEW

**Review by thefinn93 — APPROVED** [review]
:+1: LGTM

File: index.html [hunk 1]
```diff
@@ -4,6 +4,8 @@
 <title>/dev/hack</title>
 <meta name="viewport" content="width=device-width,initial-scale=1">
 <link rel="stylesheet" href="style.css">
+<script src="https://cdnjs.cloudflare.com/ajax/libs/ical.js/1.5.0/ical.min.js" integrity="sha512-0izBc1upGYnrS1u1MX7QR+sjFIxZWxLVdNI7cUoHHCutDr5ENjuQRZuS+v+3NFNGfwHSrPoHzBzED0rV651tGw==" crossorigin="anonymous" referrerpolicy="no-referrer"></script>
```
Comment (thefinn93) [inline index.html]: Can we host this ourselves? I'd rather not have 3rd party JS

Comment (leafcompost) [reply index.html]: Done!

**Comment by mjsir911** [comment]
btw the caldav export doesn't need to be proxied, can be accessed from the internet here: https://nextcloud.devhack.net/remote.php/dav/public-calendars/ecGookjEW4NtHNBi?export

**Comment by mjsir911** [comment]
Ah might spin this off to it's own ticket but we might want to specify ending times as well.

**Comment by leafcompost** [comment]
> btw the caldav export doesn't need to be proxied, can be accessed from the internet here: https://nextcloud.devhack.net/remote.php/dav/public-calendars/ecGookjEW4NtHNBi?export

Yea this is what we are using under the hood! It is just reverse proxied so we prevent any HTTP cross-origin issues without messing with NextCloud’s headers.


> Ah might spin this off to it's own ticket but we might want to specify ending times as well.

Will do in a follow up.

**Pull request merged**